reset.c

```diff
@@ -0,0 +1,888 @@
+/*  reset.c --  removes aligner updates and reference data from input sam/bam/cram file and makes read data raw
+                for new processing
+
+    Copyright (C) 2022 Genome Research Ltd.
+
+    Author: Vasudea Sarma <vasudeva.sarma@sanger.ac.uk>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE
+
+*/
+
+#include "samtools.h"
+
+#include "htslib/sam.h"
+#include "sam_opts.h"
+#include "htslib/thread_pool.h"
+#include "htslib/khash.h"
+
+KHASH_SET_INIT_INT(aux_set);        //set hold the aux tags to be retained
+KHASH_SET_INIT_STR(pg_set);         //set hold the pg tags to be removed
+
+#define TAGNUM(X) ((X)[0] << 8) | (X)[1]    //to create key for aux tags, like type key in htslib
+
+typedef struct conf_data
+{
+    int keepRGs;                    //RG line handling
+    int keepPGs;                    //PG line handling
+    khash_t(aux_set) *auxtagsin;    //SET that holds the aux tags to be reatined
+    khash_t(pg_set) *pgout;         //SET that holds the pg tags to be removed
+    int tmpfield;                   //tmp field
```

Comment:
`tmpfield` isn't very descriptive.  Given the usage, maybe rename to `reject`?

Reply:
it is not really a configuration item but a flag to select different behaviour on invocation of same method (to add aux tag or to remove aux tag in aux tag callback function, it is not relevant for pg callback). As the signature of callback is same for all and the flag is not relevant for other cases, it was made part of conf_data structure and named as 'tmp' to make it generic and usable for any purpose. Updating its comments would help?

Reply:
I'm inclined to document the current usage how it works now (and name it as such), rather than be concerned about potential future reuse.  This isn't part of the public API as it's internal to samtools rather than htslib, so we can always rename it and generalise at a later date.

Reply:
Done.